SQL assignment.txt

```diff
@@ -0,0 +1,1533 @@
+
++------+---------+-----------+------+
+| snum | sname   | city      | comm |
++------+---------+-----------+------+
+| 1001 | Peel    | London    |   12 |
+| 1002 | serres  | sanjose   |   13 |
+| 1003 | axeirod | new york  |   10 |
+| 1004 | motika  | london    |   11 |
+| 1007 | rifkin  | barcelona |   15 |
+| 1008 | fran    | london    |   25 |
++------+---------+-----------+------+
+
+
+
+32) Select all customers whose names fall in between ‘A’ and ‘G’ alphabetical range.
+
+mysql> SELECT * FROM customers WHERE cname >= "A" AND cname <="G";
+
++------+----------+---------+--------+------+
+| cnum | cname    | city    | rating | snum |
++------+----------+---------+--------+------+
+| 2006 | CLEMENS  | london  |    100 | 1001 |
+| 2008 | cisneras | sanjose |    300 | 1007 |
++------+----------+---------+--------+------+
+
+
+
+33) Select all the possible combinations of customers you can assign.
+
+mysql> SELECT C1.CNUM AS CNUM1,C1.CNAME AS CNAME1,C2.CNUM AS CNUM2,C2.CNAME AS CNAME2
+    -> FROM CUSTOMERS C1
+    -> JOIN CUSTOMERS C2
+    -> ON C1.CNUM < C2.CNUM;
+
++-------+----------+-------+----------+
+| CNUM1 | CNAME1   | CNUM2 | CNAME2   |
++-------+----------+-------+----------+
+|  2001 | hoffman  |  2002 | giovamni |
+|  2002 | giovamni |  2003 | liu      |
+|  2001 | hoffman  |  2003 | liu      |
+|  2003 | liu      |  2004 | grass    |
+|  2002 | giovamni |  2004 | grass    |
+|  2001 | hoffman  |  2004 | grass    |
+|  2004 | grass    |  2006 | CLEMENS  |
+|  2003 | liu      |  2006 | CLEMENS  |
+|  2002 | giovamni |  2006 | CLEMENS  |
+|  2001 | hoffman  |  2006 | CLEMENS  |
+|  2007 | pereira  |  2008 | cisneras |
+|  2006 | CLEMENS  |  2008 | cisneras |
+|  2004 | grass    |  2008 | cisneras |
+|  2003 | liu      |  2008 | cisneras |
+|  2002 | giovamni |  2008 | cisneras |
+|  2001 | hoffman  |  2008 | cisneras |
+|  2006 | CLEMENS  |  2007 | pereira  |
+|  2004 | grass    |  2007 | pereira  |
+|  2003 | liu      |  2007 | pereira  |
+|  2002 | giovamni |  2007 | pereira  |
+|  2001 | hoffman  |  2007 | pereira  |
++-------+----------+-------+----------+
+
+
+
+34) Select all orders that are greater than the average for October 4.
+
+mysql> SELECT * FROM ORDERS WHERE AMT > (SELECT AVG(AMT) FROM ORDERS WHERE ODATE ="10/04/90");
+
++------+---------+----------+------+------+
+| onum | amt     | odate    | cnum | snum |
++------+---------+----------+------+------+
+| 3002 |  1900.1 | 10/03/90 | 2007 | 1004 |
+| 3005 | 5160.45 | 10/03/90 | 2003 | 1002 |
+| 3006 | 1098.16 | 10/03/90 | 2008 | 1007 |
+| 3009 | 1713.23 | 10/04/90 | 2002 | 1003 |
+| 3008 |    4723 | 10/05/90 | 2006 | 1001 |
+| 3010 | 1309.95 | 10/06/90 | 2004 | 1002 |
+| 3011 | 9891.88 | 10/06/90 | 2006 | 1001 |
++------+---------+----------+------+------+
+
+
+
+35) Write a select command using correlated subquery that selects the names and numbers of all customers with ratings equal to the maximum for their city. 
+
```

Comment:
Use Correlated query as mentioned in the question.

Reply:
resolved